batch/batch/worker.py

```diff
@@ -68,6 +68,9 @@
             # 408 request timeout, 503 service unavailable
             if e.status == 408 or e.status == 503:
                 log.exception('in docker call, retrying')
+            elif e.status == 500 and ("request canceled while waiting for connection" in e.message
+                                      or "device or resource busy" in e.message):
+                log.exception('in docker call, retrying')
```

Comment:
Can we make the second one slightly more specific: ("error creating overlay mount" in e.message and "device or resource busy" in e.message)?

Reply:
I think we need "device or resource busy" in there.

These messages have 2 parts: what docker was doing (creating overlay, get http://...) and what error it got (device or resource busy, request cancelled).  It is the error that determines if it is retryable or not.  I am sure there are non-recoverable errors creating overlay mount so we definitely want to "device or resource busy".  I was just worried there might be other cases of "device or resource busy" that we don't want to retry on (or we want to be aware of), hence my suggestion to include what docker was doing.

So either go back to your original change or include both "creating overlay mount" and "device or resource busy".

Also, can you include the error full messages as comments in the text?  That's going to be useful for future reference (as we've been doing in the transient retry code).  Thanks!